legacy/reqcounter/reqcounter.go

```diff
@@ -19,33 +19,79 @@ package reqcounter
 import (
 	"fmt"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/sirupsen/logrus"
 )
 
-// RequestCounter records the number of HTTP requests to GCR.
-type RequestCounter struct {
-	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
-	requests uint64     // Number of HTTP requests since recording started.
-	since    time.Time  // When the current counter began recording requests.
+// requestCounter records the number of HTTP requests to GCR.
+// TODO: @tylerferrara in the future, add a field 'persistent bool' to determine if
+// the counter should ever reset.
+type requestCounter struct {
+	mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	requests uint64        // Number of HTTP requests since recording started.
+	since    time.Time     // When the current counter began recording requests.
+	interval time.Duration // The duration of time between each log.
 }
 
-// log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+// increment adds 1 to the request counter, signifying another call to GCR.
+func (rc *requestCounter) increment() {
+
+	atomic.AddUint64(&rc.requests, 1)
+	rc.mutex.Unlock()
```

Comment:
Actually forgot to Lock. If we didn't hold the mutex, this could happen:

1. flush grabs the mutex,
2. flush reads the counter
3. increment adds +1 to the counter
4. flush resets the counter

In this scenario, because increment did not grab the counter, we lost count of an HTTP request.